### PR TITLE
Shivani Batra Stone-paper-scissors Basics 11-8

### PR DESCRIPTION
**Pull request by batrashivani**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?1 hr

Please fill in one error and/or error message you received while working on this assignment.no output

What part of the assignment did you spend the most time on?logic

Comfort Level (1-5):4

Completeness Level (1-5):3

What did you think of this deliverable?good

Is there anything in this code that you feel pleased about?exciting

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,4 +1,29 @@
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if ((input == 'scissors') || (input == 'stone') || (input == 'paper')) {
+  var myOutputValue = getResult(input);
   return myOutputValue;
+  };
```
Comment (hpgay) [inline script.js:5]: I like how succinct the main function is!

File: script.js [hunk 2]
```diff
@@ -1,4 +1,29 @@
...
+  return "The input options are scissors, stone or paper. Kindly select from these";
 };
+
+var getResult = function(input){
+  console.log ('you have input ' + input);
+  var arrayItems = ['stone','paper','scissors'];
+  var random = Math.floor(Math.random()*3);
+  var value = arrayItems[random];
```
Comment (hpgay) [inline script.js]: Might be good to name the variables to more accurately reflect its content, in this case, e.g. var handArray, randomInteger, chosenHand etc.

File: script.js [hunk 3]
```diff
@@ -1,4 +1,29 @@
...
+    return 'System choose ' + value + ' . You won :)';
+  };
+  if ((input == 'paper') && (value == 'stone')){
+    return 'System choose ' + value + ' . You won :)';
+  };
+  if ((input == 'stone') && (value == 'scissors')){
+    return 'System choose ' + value + ' . You won:)';
+  }
+  
+  return 'System choose ' + value + ' . You lose. Try again :)';
```
Comment (hpgay) [inline script.js]: Consider grouping all the win conditions into one conditional with the use of || (or), so as to minimise repetition of code

**Review by hpgay — COMMENTED** [review]
Hi Shivani, there's good flow in your code, excellent use of arrays. Try to group the conditions together as much as possible, especially when you notice that there's repetition of code.

**Comment by batrashivani** [comment]
Hi Hsiu
Thank you so much for the detailed feedback :). Will incorporate these in
the code.
Regards

File: script.js [hunk 4]
```diff
@@ -1,4 +1,29 @@
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if ((input == 'scissors') || (input == 'stone') || (input == 'paper')) {
+  var myOutputValue = getResult(input);
```
[the remark this answers is not in the document]
Comment (batrashivani) [reply script.js:3]: Do I have to declare all functions referenced from main on top of the script.js? The code is running fine even if I declare the functions after main. Please let me know. Thanks.

File: script.js [hunk 5]
```diff
@@ -1,4 +1,24 @@
...
   return myOutputValue;
+  };
+  return "The input options are scissors, stone or paper. Kindly select from these";
 };
+
+var getResult = function(input){
+  console.log ('you have input ' + input);
```
[the remark this answers is not in the document]
Comment (batrashivani) [reply script.js:10]: Sorry, I didn't get this? Where exactly is the extra space between function name and parenthesis? Please let me know.

**Comment by batrashivani** [comment]
Fixed code formatting.

**Pull request closed without merging**